### PR TITLE
Polaris - OMF API Python - URL manipulation

### PR DESCRIPTION
**Pull request by cbardini**
225109

### CODE REVIEW

**Review by derekendres — APPROVED** [review]

File: program.py [hunk 1]
```diff
@@ -61,29 +61,29 @@ def get_token(endpoint):
 
     # we can't short circuit it, so we must go retreive it.
 
-    discoveryUrl = requests.get(
+    discovery_URL = requests.get(
```
Comment (kelseyduffy) [inline program.py:64]: is capitalizing 'URL' proper snake_case?

File: test.py [hunk 2]
```diff
@@ -147,28 +148,33 @@ def send_get_request_to_endpoint(endpoint, path='', base=''):
     msg_headers.pop('omfversion')
     msg_headers["Accept-Verbosity"] = 'verbose'
 
+    # construct and validate url
+    url = urlparse(base+path)
```
Comment (kelseyduffy) [inline test.py:152]: should there be spaces around the '+'?

**Review by kelseyduffy — APPROVED** [review]

**Comment by kelseyduffy** [comment]
I approved it but didn't merge yet so you have a chance to apply the styling comments if you agree with them

File: program.py (hunk 1 above)
Comment (cbardini) [reply program.py]: I changed 'URL' to lower case

File: program.py [hunk 3]
```diff
@@ -61,29 +61,29 @@ def get_token(endpoint):
...
-        discoveryUrl.close()
-        raise Exception(f'Failed to get access token endpoint from discovery URL: {discoveryUrl.status_code}:{discoveryUrl.text}')
+    if discovery_URL.status_code < 200 or discovery_URL.status_code >= 300:
+        discovery_URL.close()
+        raise Exception(f'Failed to get access token endpoint from discovery URL: {discovery_URL.status_code}:{discovery_URL.text}')
 
-    tokenEndpoint = json.loads(discoveryUrl.content)["token_endpoint"]
-    tokenUrl = urlparse(tokenEndpoint)
+    token_endpoint = json.loads(discovery_URL.content)["token_endpoint"]
+    token_URL = urlparse(token_endpoint)
```
[the remark this answers is not in the document]
Comment (cbardini) [reply program.py]: I changed 'URL' to lower case

File: test.py (hunk 2 above)
Comment (cbardini) [reply test.py]: Not sure, but I suppose it doesn't hurt so I added some

**Pull request merged**